diff --git a/api.mjs b/api.mjs
index df1c48e05be38829e97f175445c374ab28334dc4..8e3784074418d89256cdc3783fe8bdaaa0528099 100644
--- a/api.mjs
+++ b/api.mjs
@@ -1,3 +1,9 @@
+// Allows drizzle-kit/api to be used in Node ESM context.
+// See issue: https://github.com/drizzle-team/drizzle-orm/issues/2853
+// Similar to this fix: https://github.com/hey-api/openapi-ts/pull/1086/files
+import { createRequire } from 'module'
+const require = createRequire(import.meta.url)
+
 var __create = Object.create;
 var __defProp = Object.defineProperty;
 var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
